### PR TITLE
Fix VMware memory retrieval

### PR DESCRIPTION
**Pull request by JoaoJandre**
### Description

Currently memory usage statistics in VMware are inverted. It was discovered that during statistics collection, memory usage was collected and passed as free memory, so the memory usage graphs were inverted. The free memory calculation has been fixed and the memory statistics are now consistent with what is observed in VMware.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?
This was tested in a local lab, by creating a VM with VMware as its hypervisor and checking memory statistics in both vCenter and ACS.

### CODE REVIEW

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 1]
```diff
@@ -6535,7 +6535,10 @@ private HashMap<String, VmStatsEntry> getVmStats(List<String> vmNames) throws Ex
                         }
                     }
 
-                    final VmStatsEntry vmStats = new VmStatsEntry(0, NumberUtils.toDouble(memkb) * 1024, NumberUtils.toDouble(guestMemusage) * 1024, NumberUtils.toDouble(memlimit) * 1024,
+                    double doubleMemKb = NumberUtils.toDouble(memkb)
+                    double guestFreeMem =  doubleMemKb - NumberUtils.toDouble(guestMemusage);
+
+                    final VmStatsEntry vmStats = new VmStatsEntry(0, doubleMemKb * 1024, guestFreeMem * 1024, NumberUtils.toDouble(memlimit) * 1024, NumberUtils.toDouble(memlimit) * 1024,
```
Comment (DaanHoogland) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:6541]: if I count correctly, you are replacing 4 for 5 parameters to the constructor on VmStatsEntry. It takes 13 and now has 14 in the call. I don´t see a change to the constructor itself.

Comment (JoaoJandre) [reply plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: I'm sorry, I probably made a mistake when cherry-picking this change.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by nvazquez** [comment]
Hi @JoaoJandre can you please target this PR to the 4.17 branch?

**Review by GutoVeronezi — APPROVED** [review]
LGTM

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by sureshanaparti** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test centos7 vmware-67u3

**Review by nvazquez — APPROVED** [review]

**Pull request merged**